aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteCommandFactory.java

```diff
@@ -425,9 +447,17 @@ private List<Object> extractFieldValues(@NonNull Model model) throws DataStoreEx
                 new SQLiteModelFieldTypeConverter(schema, schemaRegistry, gson);
         final Map<String, ModelField> modelFields = schema.getFields();
         final List<Object> bindings = new ArrayList<>();
+        Object fieldValue;
         for (SQLiteColumn column : table.getSortedColumns()) {
-            final ModelField modelField = Objects.requireNonNull(modelFields.get(column.getFieldName()));
-            final Object fieldValue = converter.convertValueFromTarget(model, modelField);
+            if (column.getName().equals(SQLiteTable.PRIMARY_KEY_FIELD_NAME)) {
+                fieldValue = model.getPrimaryKeyString();
+            } else if (column.isForeignKey()) {
+                final ModelField modelField = Objects.requireNonNull(modelFields.get(column.getOwnedField()));
```

Comment:
This line throws NPE for the following schema when saving the child model in this belongs to relationship.

```graphql
type BelongsToParent @model {
  id: ID!
  name: String
  explicitChild: BelongsToChildExplicit @hasOne
}

type BelongsToChildExplicit @model {
  id: ID!
  name: String
  belongsToParentID: ID
  belongsToParent: BelongsToParent @belongsTo(fields: ["belongsToParentID"])
}
```

This line resolves column name as `belongsToParentID` which is correct. The codegen generated model, however, doesn't list this filed as a model field as it's used as belongs to connection field (which makes this filed hidden from customers). Hence, `modelFields.get(column.getOwnedField())` returns null.

Codegen generated Dart model class
<img width="578" alt="image" src="https://user-images.githubusercontent.com/10602282/169925924-29334c1d-84f8-4c2f-81e5-ca224213559b.png">

Codegen generated Java model class
<img width="1294" alt="image" src="https://user-images.githubusercontent.com/10602282/169925992-b105fe5a-1a0f-406d-9bb8-9dcb459b047d.png">

Reply:
I might have fixed this already, creating the PR for that right now.

Reply:
Can you please test it again?